RecoLocalCalo/HcalRecProducers/plugins/alpaka/HBHERecHitProducerPortable.cc

```diff
@@ -0,0 +1,169 @@
+    configParameters_.firstSampleShift = ps.getParameter<int>("firstSampleShift");
+    //TODO: produce only pedestals_width or convertedPedestalWidths depending on this bool
+    configParameters_.useEffectivePedestals = ps.getParameter<bool>("useEffectivePedestals");
+
+    configParameters_.meanTime = ps.getParameter<double>("meanTime");
+    configParameters_.timeSigmaSiPM = ps.getParameter<double>("timeSigmaSiPM");
+    configParameters_.timeSigmaHPD = ps.getParameter<double>("timeSigmaHPD");
+    configParameters_.ts4Thresh = ps.getParameter<double>("ts4Thresh");
+
+    configParameters_.applyTimeSlew = ps.getParameter<bool>("applyTimeSlew");
+    auto const tzeroValues = ps.getParameter<std::vector<double>>("tzeroTimeSlewParameters");
+    auto const slopeValues = ps.getParameter<std::vector<double>>("slopeTimeSlewParameters");
+    auto const tmaxValues = ps.getParameter<std::vector<double>>("tmaxTimeSlewParameters");
+
+    configParameters_.tzeroTimeSlew = tzeroValues[HcalTimeSlew::Medium];
+    configParameters_.slopeTimeSlew = slopeValues[HcalTimeSlew::Medium];
+    configParameters_.tmaxTimeSlew = tmaxValues[HcalTimeSlew::Medium];
+
+    auto threadsMinimize = ps.getParameter<std::vector<uint32_t>>("kernelMinimizeThreads");
+    configParameters_.kernelMinimizeThreads[0] = threadsMinimize[0];
+    configParameters_.kernelMinimizeThreads[1] = threadsMinimize[1];
+    configParameters_.kernelMinimizeThreads[2] = threadsMinimize[2];
+  }
+
+  HBHERecHitProducerPortable::~HBHERecHitProducerPortable() {}
+
+  void HBHERecHitProducerPortable::fillDescriptions(edm::ConfigurationDescriptions& cdesc) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::ESInputTag>("mahiPulseOffSets", edm::ESInputTag(""));
+    desc.add<uint32_t>("maxTimeSamples", 10);
+    desc.add<uint32_t>("kprep1dChannelsPerBlock", 32);
+    desc.add<edm::InputTag>("digisLabelF01HE", edm::InputTag{"hcalRawToDigiGPU", "f01HEDigisGPU"});
+    desc.add<edm::InputTag>("digisLabelF5HB", edm::InputTag{"hcalRawToDigiGPU", "f5HBDigisGPU"});
+    desc.add<edm::InputTag>("digisLabelF3HB", edm::InputTag{"hcalRawToDigiGPU", "f3HBDigisGPU"});
+    desc.add<std::string>("recHitsLabelM0HBHE", "recHitsM0HBHE");
+    desc.add<int>("sipmQTSShift", 0);
+    desc.add<int>("sipmQNTStoSum", 3);
+    desc.add<int>("firstSampleShift", 0);
+    desc.add<bool>("useEffectivePedestals", true);
+
+    desc.add<double>("meanTime", 0.f);
+    desc.add<double>("timeSigmaSiPM", 2.5f);
+    desc.add<double>("timeSigmaHPD", 5.0f);
+    desc.add<double>("ts4Thresh", 0.0);
+
+    desc.add<bool>("applyTimeSlew", true);
+    desc.add<std::vector<double>>("tzeroTimeSlewParameters", {23.960177, 11.977461, 9.109694});
+    desc.add<std::vector<double>>("slopeTimeSlewParameters", {-3.178648, -1.5610227, -1.075824});
+    desc.add<std::vector<double>>("tmaxTimeSlewParameters", {16.00, 10.00, 6.25});
+    desc.add<std::vector<uint32_t>>("kernelMinimizeThreads", {16, 1, 1});
```

Comment:
Reading the code, I think only the first element is used.
Can this be changed to a simple `uint32_t` ?

Reply:
This follows from the current MAHI implementation. 
Maybe HCAL DPG can comment if there are reason to keep the full vector? @mariadalfonso